### ISSUE TITLE
[2.2][hostname module] Could not set property: Failed to activate service 'org.freedesktop.hostname1': timed out

### ISSUE THREAD

**Issue by juju4** [ansible/ansible#19814]
_From @juju4 on November 11, 2016 23:34_

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
hostname module

##### ANSIBLE VERSION
```
# ansible --version
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT
local run on ubuntu xenial (inside lxc containers with kitchen-test+kitchen-ansible)

##### SUMMARY
hostname module fails to set with above error

##### STEPS TO REPRODUCE
```
- debug: var=hostname_hostname
- name: set hostname
  hostname: "name={{ hostname_hostname }}"
```

##### EXPECTED RESULTS
set hostname without error.
was working with ansible 2.1

##### ACTUAL RESULTS
task is failing

```
TASK [hostname : debug] ********************************************************
task path: /tmp/kitchen/hostname/tasks/main.yml:3
ok: [localhost] => {
    "hostname_hostname": "example"
}

TASK [hostname : set hostname] *************************************************
task path: /tmp/kitchen/hostname/tasks/main.yml:4
Using module file /usr/local/lib/python2.7/dist-packages/ansible/modules/core/system/hostname.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250 `" && echo ansible-tmp-1478906878.17-90221180289250="` echo $HOME/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250 `" ) && sleep 0'
<localhost> PUT /tmp/tmpDdbhyS TO /root/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250/hostname.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250/ /root/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250/hostname.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250/hostname.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1478906878.17-90221180289250/" > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "name": "example"
        }, 
        "module_name": "hostname"
    }, 
    "msg": "Command failed rc=1, out=, err=Could not set property: Failed to activate service 'org.freedesktop.hostname1': timed out\n"
}
        to retry, use: --limit @/tmp/kitchen/default.retry

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1 
```

trying 
```
systemctl restart systemd-logind
```
and restarting playbook doesn't help

using hostname command directly is working.

_Copied from original issue: ansible/ansible-modules-core#5582_

**Comment by juju4** [comment]
_From @ansibot on November 11, 2016 23:34_



@alikins, @saito-hideki, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by juju4** [comment]
_From @saito-hideki on November 14, 2016 8:47_

I reproduced this issue.

If your target linux system is managed systemd, current hostname module uses hostnamectl to manage hostname.
But, hostnamectl fails under lxd unpriv container(1). So now, hostname module can't manage this environment until fixing this bug.

Additional information:
1. https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1575779

**Comment by juju4** [comment]
Ok. Thanks @saito-hideki

**Comment by juju4** [comment]
_From @wvangeit on November 18, 2016 14:50_

+1

**Comment by juju4** [comment]
_From @ansibot on December 7, 2016 20:4_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by AJamesPhillips** [comment]
Can I help with fixing this?  I'm trying to change the hostname on a raspberry pi from raspberrypi to something else and have the same error:  "Command failed rc=1, out=, err=Could not set property: The name org.freedesktop.PolicyKit1 was not provided by any .service files"

**Comment by soloradish** [comment]
I encountered this issue in centos 7 kvm guest.

The problem was gone, after I turned off selinux on kvm guest.

**Comment by Vedrillan** [comment]
This still affects ansible 2.5.1 on unprivileged lxc.

**Comment by x-yuri** [comment]
`hostnamectl` might timeout when invoked e.g. inside Ubuntu Xenial LXC container running on Arch Linux host (kernel-4.19.4). Changing `PrivateDevices` in `/lib/systemd/system/systemd-hostnamed.service` to `no` [resolves][1] the issue.

[1]: https://discuss.linuxcontainers.org/t/systemd-hostnamed-service-failed-at-step-namespace-spawning-lib-systemd-systemd-hostnamed-invalid-argument/3740

Also, it doesn't work when `dbus` is not installed. That's the case in a freshly created Ubuntu Xenial LXC container.

**Comment by dachshund-digital** [comment]
So is no one actually working on resolving this issue?  This issue is really causing a problem for me as well.

**Comment by x-yuri** [comment]
@dachshund-digital 
> Changing PrivateDevices in /lib/systemd/system/systemd-hostnamed.service to no resolves the issue.

**Comment by dachshund-digital** [comment]
Unfortunately, that did not seem to work.  I still get the error that host name can not be changed when systemd-networkd attempts it.  States that authentication is required.  This happens on Debian 9 stretch as well Raspbian OS (Raspberry Pi Debian 9 Stretch for ARM).

**Comment by x-yuri** [comment]
@dachshund-digital I do not follow. You're provisioning a physical server or an LXC container? The solution I was referring to is for LXC containers apparently. What does it matter which NICs it has, DHCP and all? By log you mean systemd log? What are the exact messages?

> Host name defaults to localhost

You try to change hostname, but it instead changes to localhost? Are you sure you're facing the same problem as the one described in the first post?

**Comment by dachshund-digital** [comment]
I am doing all of the above... physical provisioning, virtual container provision, I am even working on methods to mass deploy micro-controllers like the Raspberry Pi.  This a variant of the same problem, being able to drive hostname change from a logical and efficient method driven by DHCP.  The change to localhost is a side effect of how Raspbian/Debian default during an unattended/preseed installation sequence.  The expectation was to use ansible to _correct_ the issue post initial deployment.  The fact that Debian/Raspian has the dumbest preseed process is a separate issue, no other distribution has as much trouble setting the hostname correctly from a DHCP source as Debian.  The preseed process works well, except for this one quirk with hostname provisioning, and now that ansible has a comparable issue, frankly, it is frustrating.  This just should not as big an issue as it (apparently) is.

**Comment by mikerev** [comment]
Bump. hostname functionality is a day 1 feature.

**Comment by x-yuri** [comment]
@mikerev You can contribute by looking into why `hostnamectl` doesn't work on your server.

**Comment by mikerev** [comment]
> @mikerev You can contribute by looking into why `hostnamectl` doesn't work on your server.

Will do

**Comment by nielspeen** [comment]
I ran into this issue after switching from _ifupdown_ (Ubuntu) to _netplan_.io with _networkd_ as the renderer.

Apparently `hostnamectl` depends on dbus, but neither systemd nor netpan.io list it as a dependency.

`sudo apt install dbus` fixed the issue for me.

**Comment by x-yuri** [comment]
That's [one][1] of the reasons.

[1]: https://github.com/ansible/ansible/issues/19814#issuecomment-452726188

**Comment by cristichiru** [comment]
Had the same issue on Ubuntu 18.04.2 and was fixed with **rebooting the machines**.

Everything worked until I have re-run  kubespray a couple of times, eventually hitting the lock.

I had dbus installed. Tried adding `/etc/systemd/system/systemd-hostnamed.service.d/override.conf`
```ini
[Service]
PrivateNetwork=no
```
Restarted, bu `systemd-hostnamed` still hanged.

Searching around, led me to this thread and:
- https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1780227
- https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1575779

**Comment by lijinghuatongxue** [comment]
my roles file,it work ！

- hostname: name={{ HOST_NAME }}-{{ ansible_default_ipv4.address.split('.')[-2] }}-{{ ansible_default_ipv4.address.split('.')[3]  }}
  when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
  sudo: yes



output：

TASK [host-init : Print HostkName] ****************************************************************
ok: [172.29.207.38] => {
    "msg": "New hostname Set as AWSNX-RC-QA-Test-lijinghua-207-38"
}

**Comment by bmalynovytch** [comment]
> Had the same issue on Ubuntu 18.04.2 and was fixed with **rebooting the machines**.
> 
> Everything worked until ...

I can confirm that rebooting solves the issue for a while.
In my case, everything worked in local, they started to fail when accessing the host through an OpenVPN tunnel.
```
failing host <=== OpenVPN link ===> OpenVPN server <=== OpenVPN link ===> laptop with Ansible
```

**Comment by mvernimmen** [comment]
A reboot worked around this problem for me with ansible 2.9.1 on CentOS Linux release 7.7.1908 (Core) with kernel 3.10.0-1062.4.3.el7.x86_64

**Comment by s-hertel** [comment]
Closing as a duplicate of #19681.
